src/internal/observable/combineLatest.ts

```diff
@@ -11,322 +11,45 @@ import { popResultSelector, popScheduler } from '../util/args';
->(
-  v1: O1,
-  v2: O2,
-  v3: O3,
-  v4: O4,
-  v5: O5,
-  v6: O6,
-  resultSelector: (
-    v1: ObservedValueOf<O1>,
-    v2: ObservedValueOf<O2>,
-    v3: ObservedValueOf<O3>,
-    v4: ObservedValueOf<O4>,
-    v5: ObservedValueOf<O5>,
-    v6: ObservedValueOf<O6>
-  ) => R,
-  scheduler?: SchedulerLike
-): Observable<R>;
 
-// With a scheduler (deprecated)
-/** @deprecated The scheduler argument is deprecated, use scheduled and combineAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
-export function combineLatest<O1 extends ObservableInput<any>>(sources: [O1], scheduler: SchedulerLike): Observable<[ObservedValueOf<O1>]>;
-/** @deprecated The scheduler argument is deprecated, use scheduled and combineAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
-export function combineLatest<O1 extends ObservableInput<any>, O2 extends ObservableInput<any>>(
-  sources: [O1, O2],
-  scheduler: SchedulerLike
-): Observable<[ObservedValueOf<O1>, ObservedValueOf<O2>]>;
-/** @deprecated The scheduler argument is deprecated, use scheduled and combineAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
-export function combineLatest<O1 extends ObservableInput<any>, O2 extends ObservableInput<any>, O3 extends ObservableInput<any>>(
-  sources: [O1, O2, O3],
-  scheduler: SchedulerLike
-): Observable<[ObservedValueOf<O1>, ObservedValueOf<O2>, ObservedValueOf<O3>]>;
-/** @deprecated The scheduler argument is deprecated, use scheduled and combineAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
-export function combineLatest<
-  O1 extends ObservableInput<any>,
-  O2 extends ObservableInput<any>,
-  O3 extends ObservableInput<any>,
-  O4 extends ObservableInput<any>
->(
-  sources: [O1, O2, O3, O4],
-  scheduler: SchedulerLike
-): Observable<[ObservedValueOf<O1>, ObservedValueOf<O2>, ObservedValueOf<O3>, ObservedValueOf<O4>]>;
-/** @deprecated The scheduler argument is deprecated, use scheduled and combineAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
-export function combineLatest<
-  O1 extends ObservableInput<any>,
-  O2 extends ObservableInput<any>,
-  O3 extends ObservableInput<any>,
-  O4 extends ObservableInput<any>,
-  O5 extends ObservableInput<any>
->(
-  sources: [O1, O2, O3, O4, O5],
-  scheduler: SchedulerLike
-): Observable<[ObservedValueOf<O1>, ObservedValueOf<O2>, ObservedValueOf<O3>, ObservedValueOf<O4>, ObservedValueOf<O5>]>;
 /** @deprecated The scheduler argument is deprecated, use scheduled and combineAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
-export function combineLatest<
-  O1 extends ObservableInput<any>,
-  O2 extends ObservableInput<any>,
-  O3 extends ObservableInput<any>,
-  O4 extends ObservableInput<any>,
-  O5 extends ObservableInput<any>,
-  O6 extends ObservableInput<any>
->(
-  sources: [O1, O2, O3, O4, O5, O6],
+export function combineLatest<A extends readonly unknown[], R>(
+  sources: readonly [...ObservableInputTuple<A>],
```

Comment:
issue: `R` is not used in this sig. The return type should be `Observable<A>`, AFAICT.

nitpick: Why is the surrounding `[... ]` needed? Why not `sources: ObservableInputTuple<A>`? This occurs elsewhere, too.

Reply:
Having used this, I can see, now, that the `[... ]` business is needed.